persistent/test/Database/Persist/TH/KindEntitiesSpecImports.hs

```diff
@@ -0,0 +1,21 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE LambdaCase #-}
+
+module Database.Persist.TH.KindEntitiesSpecImports where
+
+import qualified Data.Text as T
+import TemplateTestImports
+
+data Owner = MerchantOwned | CustomerOwned
+data AccountKind = Debit | Credit
+
+newtype MoneyAmount (a :: Owner) (b :: AccountKind) = MoneyAmount Rational
+
+instance PersistFieldSql (MoneyAmount a b) where
+    sqlType _ = SqlInt64
+
+instance PersistField (MoneyAmount a b) where
+    toPersistValue (MoneyAmount n) = PersistRational n
```

Comment:
I don't think this `PeristRational` lines up with the `SqlInt64` type you specified